### ISSUE TITLE
Algolia search modal under HomepageHeader component in production build

### ISSUE THREAD

**Issue by stnor** [facebook/docusaurus#4941]
### Description
The Algolia search dialog is presented under the HomepageHeader component (in production builds only).

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes
## To Reproduce

See https://nomp-info.se/

I "fixed" the issue by adding z-index: -1 on the component.

## Expected behavior

I would expect the search box to be placed on top of the HomepageHeader.

## Actual Behavior

<img width="1163" alt="Screenshot 2021-06-10 at 14 14 47" src="https://user-images.githubusercontent.com/561167/121523355-6810c780-c9f6-11eb-9269-293833ee8658.png">

## Your Environment

Chrome latest.

- Public site URL: https://nomp-info.se/
- Docusaurus version used: beta.0
- Environment name and version: node v15.0.1, Chrome 91.0.4472.77
- Operating system and version: macOS

```
  "dependencies": {
    "@docusaurus/core": "2.0.0-beta.0",
    "@docusaurus/plugin-sitemap": "2.0.0-beta.0",
    "@docusaurus/preset-classic": "2.0.0-beta.0",
    "@mdx-js/react": "^1.6.21",
    "@svgr/webpack": "^5.5.0",
    "clsx": "^1.1.1",
    "file-loader": "^6.2.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "rss-to-json": "^1.1.3",
    "url-loader": "^4.1.1"
  },
```
## Reproducible Demo
See https://nomp-info.se/

**Comment by slorber** [comment]
You are just unlucky and affected by a temporary cssnano@5.0.3 bug that is solved in 5.0.4

Deleting your lock file and regenerating will pick a new version of cssnano and likely fix the issue

You can notice it on your layout also:

![image](https://user-images.githubusercontent.com/749374/121530252-8928e680-c9fd-11eb-916f-c5ffd57e441c.png)

**Comment by stnor** [comment]
Thanks. Awesome!
